identity/aziot-hub-client-async/src/lib.rs

```diff
@@ -159,7 +159,7 @@ impl Client {
         TRequest: serde::Serialize,
         TResponse: serde::de::DeserializeOwned,
     {
-        let uri = format!("https://{}{}", hub_device.iothub_hostname, uri);
+        let uri = format!("https://{}{}", hub_device.local_gateway_hostname, uri);
```

Comment:
https://github.com/Azure/iot-identity-service/blob/main/identity/aziot-hub-client-async/src/lib.rs#L311-L321
Needs to be changed in `request_no_content` too